examples/linear_model/plot_sparse_logistic_regression_20newsgroups.py

```diff
@@ -82,6 +82,7 @@
             penalty="l1",
             max_iter=this_max_iter,
             random_state=42,
+            warm_start=True,
```

Comment:
you sure this has an effect? I'm not sure if in this case it should

Reply:
I am not so deep in how exactly they run these modules/their tests but this would have an influence depending on how often they are executed and in what order (e.g. multiple runs)

Reply:
Since you are creating a new `lr` instance each time you call fit, it should not have any impact. 


`warm_start=True` could have an impact if you move the creation of the `lr` instance before entering the `for this_max_iter in model_params["iters"]` loop and then only calling the `lr.set_params(max_iter=this_max_iter)` inside the inner loop prior to calling fit. However wall time measurements would get a different meaning and would need to be readjusted before plotting. I have the feeling that this be too tricky to achieve or would render the example too complex.

Let's just use the training set size reduction and revert the use of `warm_start=True`.

Reply:
@ogrisel Will do